### PR TITLE
LCM spring boot version

### PR DESCRIPTION
**Pull request by MrSebastian**
**Description:**  

LCM of the spring boot version
- 3.0.7 -> 3.1.5 -> 3.2.1
- update of plugin versions
- fix of one cve (commons-test)

**Reference:**

Issue: #204 

Notifying additional team members:

@mirrodi @AnHo314

---
**Definition of Done (DoD):**

- [x] Unittests gepflegt - nothing todo
- [x] Dokumentation ist gepflegt - nothing todo

### CODE REVIEW

**Comment by MrSebastian** [comment]
> I could not start the backend service. It seems like the query method "findAllUnassignedInSpecificMeldzeitraum" in the NwkRepository has some issues with this update

After building the jar file and running with profile `local` I get the following (shortened) stacktrace:
```
[...]
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'nwkRepository' defined in de.muenchen.oss.praktikumsplaner.repository.NwkRepository defined in @EnableJpaRepositories declared on MicroServiceApplication: Could not create query for public abstract java.util.List de.muenchen.oss.praktikumsplaner.repository.NwkRepository.findAllUnassignedInSpecificMeldzeitraum(java.util.UUID); Reason: Validation failed for query for method public abstract java.util.List de.muenchen.oss.praktikumsplaner.repository.NwkRepository.findAllUnassignedInSpecificMeldzeitraum(java.util.UUID)
Caused by: org.springframework.data.repository.query.QueryCreationException: Could not create query for public abstract java.util.List de.muenchen.oss.praktikumsplaner.repository.NwkRepository.findAllUnassignedInSpecificMeldzeitraum(java.util.UUID); Reason: Validation failed for query for method public abstract java.util.List de.muenchen.oss.praktikumsplaner.repository.NwkRepository.findAllUnassignedInSpecificMeldzeitraum(java.util.UUID)
Caused by: java.lang.IllegalArgumentException: Validation failed for query for method public abstract java.util.List de.muenchen.oss.praktikumsplaner.repository.NwkRepository.findAllUnassignedInSpecificMeldzeitraum(java.util.UUID)
Caused by: java.lang.IllegalArgumentException: org.hibernate.query.SemanticException: Cannot compare left expression of type 'java.util.UUID' with right expression of type 'de.muenchen.oss.praktikumsplaner.domain.Nwk'
Caused by: org.hibernate.query.SemanticException: Cannot compare left expression of type 'java.util.UUID' with right expression of type 'de.muenchen.oss.praktikumsplaner.domain.Nwk'
```

**Comment by MrSebastian** [comment]
> I could not start the backend service. It seems like the query method "findAllUnassignedInSpecificMeldzeitraum" in the NwkRepository has some issues with this update

@LuisEinig 
ec41e938823214b03b39815f628914551dcc9a91 should fix it

---

this problem reveals that we have some trouble with our testcoverage

**Comment by LuisEinig** [comment]
grafik
grafik
"Name for argument of type [java.lang.String] not specified, and parameter name information not available via reflection. Ensure that the compiler uses the '-parameters' flag."

When opening the `Praktikumsplätze ` page, the request throws the error above.

**Comment by MrSebastian** [comment]
> grafik grafik "Name for argument of type [java.lang.String] not specified, and parameter name information not available via reflection. Ensure that the compiler uses the '-parameters' flag."
> 
> When opening the `Praktikumsplätze ` page, the request throws the error above.

resolved with [31352e7](https://github.com/it-at-m/Praktikumsplaner/pull/200/commits/31352e79000f18d20a5e361f5b96bc009dc08ff6)

**Comment by LuisEinig** [comment]
"Name for argument of type [java.util.UUID] not specified, and parameter name information not available via reflection. Ensure that the compiler uses the '-parameters' flag."
same error appears when trying to assign and unassign a Nwk.

**Comment by MrSebastian** [comment]
> "Name for argument of type [java.util.UUID] not specified, and parameter name information not available via reflection. Ensure that the compiler uses the '-parameters' flag." same error appears when trying to assign and unassign a Nwk.

fixed mit [377fad1](https://github.com/it-at-m/Praktikumsplaner/pull/200/commits/377fad1d377cb0a0701cc190bdc5d5ce7e3ae703)

File: praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/rest/PraktikumsstellenController.java [hunk 1]
```diff
@@ -81,7 +81,7 @@ public TreeMap<String, List<PraktikumsstelleDto>> getAllPraktiumsstellenInSpecif
     @PreAuthorize("hasRole('ROLE_' + T(de.muenchen.oss.praktikumsplaner.security.AuthoritiesEnum).AUSBILDUNGSLEITUNG.name())")
     @PatchMapping("/{praktikumsstellenId}")
     @ResponseStatus(HttpStatus.OK)
-    public PraktikumsstelleDto assignNwk(@PathVariable UUID praktikumsstellenId, @RequestParam(required = false) UUID nwkId) {
+    public PraktikumsstelleDto assignNwk(@PathVariable UUID praktikumsstellenId, @RequestParam(name = "nwkId", required = false) UUID nwkId) {
```
Comment (LuisEinig) [inline praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/rest/PraktikumsstellenController.java:84]:
```suggestion
    public PraktikumsstelleDto assignNwk(@PathVariable(name = "praktikumsstellenId") UUID praktikumsstellenId,
            @RequestParam(name = "nwkId", required = false) UUID nwkId) {
```
Seems like this is needed to fix it completely

**Review by LuisEinig — APPROVED** [review]

**Pull request merged**